src/simplewallet/simplewallet.cpp

```diff
@@ -412,6 +423,521 @@ bool simple_wallet::payment_id(const std::vector<std::string> &args/* = std::vec
+    try
+    {
+      std::string body(data, headerlen);
+      std::stringstream iss;
+      iss << body;
+      std::vector<tools::wallet2::multisig_info> i;
+      try
+      {
+        boost::archive::portable_binary_iarchive ar(iss);
+        ar >> i;
+      }
+      catch (...)
+      {
+        iss.str("");
+        iss << body;
+        boost::archive::binary_iarchive ar(iss);
+        ar >> i;
+      }
+      message_writer() << boost::lexical_cast<std::string>(i.size()) << " outputs found in " << filename;
+      info.push_back(std::move(i));
+    }
+    catch (const std::exception &e)
+    {
+      fail_msg_writer() << "Failed to import multisig info: " << e.what();
+      return true;
+    }
+  }
+
+  LOCK_IDLE_SCOPE();
+
+  // all read and parsed, actually import
+  try
+  {
+    size_t n_outputs = m_wallet->import_multisig(info);
+    success_msg_writer() << "Multisig info imported";
+  }
+  catch (const std::exception &e)
+  {
+    fail_msg_writer() << "Failed to import multisig info: " << e.what();
+    return true;
+  }
+  if (m_trusted_daemon)
+  {
+    try
+    {
+      m_wallet->rescan_spent();
+    }
+    catch (const std::exception &e)
+    {
+      message_writer() << "Failed to update spent status after importing multisig info: " << e.what();
+    }
+  }
+  else
+  {
+    message_writer() << "Untrusted daemon, spent status may be incorrect. Use a trusted daemon and run \"rescan_spent\"";
+  }
+  return true;
+}
+
+bool simple_wallet::accept_loaded_tx(const tools::wallet2::multisig_tx_set &txs)
+{
+  std::string extra_message;
+  return accept_loaded_tx([&txs](){return txs.m_ptx.size();}, [&txs](size_t n)->const tools::wallet2::tx_construction_data&{return txs.m_ptx[n].construction_data;}, extra_message);
+}
+
+bool simple_wallet::sign_multisig(const std::vector<std::string> &args)
+{
+  if(!m_wallet->multisig())
+  {
+    fail_msg_writer() << tr("This is not a multisig wallet");
+    return true;
+  }
+  if (args.size() != 1)
+  {
+    fail_msg_writer() << tr("usage: sign_multisig <filename>");
+    return true;
+  }
+  if (m_wallet->ask_password() && !get_and_verify_password()) { return true; }
+
+  std::string filename = args[0];
+  std::vector<crypto::hash> txids;
+  uint32_t signers = 0;
+  try
+  {
+    bool r = m_wallet->sign_multisig_tx(filename, txids, [&](const tools::wallet2::multisig_tx_set &tx){ signers = tx.m_signers.size(); return accept_loaded_tx(tx); });
+    if (!r)
+    {
+      fail_msg_writer() << tr("Failed to sign multisig transaction");
+      return true;
+    }
+  }
+  catch (const tools::error::multisig_export_needed& e)
+  {
+    fail_msg_writer() << tr("Multisig error: ") << e.what();
+    return true;
+  }
+  catch (const std::exception &e)
+  {
+    fail_msg_writer() << tr("Failed to sign multisig transaction: ") << e.what();
+    return true;
+  }
+
+  if (txids.empty())
+  {
+    uint32_t threshold;
+    m_wallet->multisig(&threshold);
+    uint32_t signers_needed = threshold - signers - 1;
```

Comment:
>return sign_multisig_tx(exported_txs, filename, txids);
>exported_txs.m_signers.insert(hash);

**Thank you** for clarifying 😄! I hadn't looked closer beyond those checks (like I said on IRC, I had only gave a once-over review).